Lib/typing.py

```diff
@@ -1393,7 +1393,11 @@ def __mro_entries__(self, bases):
                 return ()
             i = bases.index(self)
             for b in bases[i+1:]:
-                if isinstance(b, _BaseGenericAlias) and b is not self:
+                if b is self:
+                    continue
+                if isinstance(b, _BaseGenericAlias):
```

Comment:
This fix would be fine for me for now, at least until we make `typing._GenericAlias` a subclass of `types.GenericAlias` (as proposed in #103721), which would be a much larger and riskier change:

```suggestion
                if isinstance(b, (_BaseGenericAlias, types.GenericAlias)):
```